subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/IncrementalResultStoringCompiler.java

```diff
@@ -105,4 +108,21 @@ private AnnotationProcessingData convertProcessingResult(AnnotationProcessingRes
         }
         return result;
     }
+
+    private Set<GeneratedResource> internResources(Set<GeneratedResource> resources) {
+        Set<GeneratedResource> result = Sets.newHashSet();
+        for (GeneratedResource r : resources) {
+            // TODO: is not interning the GeneratedResources themselves a problem?
+            result.add(new GeneratedResource(r.getLocation(), interner.intern(r.getPath())));
+        }
+        return result;
+    }
```

Comment:
The question in the TODO stands.

Reply:
Interning resources isn't necessary, as they are never referenced again. We intern class names because they can reference each other, so we save a lot of space on those references.

Reply:
Got it.  Do you think I should still intern the path?  Does it matter one way or the other?

Reply:
No, that's what I meant - Interning the path will not help and only cost CPU cycles.

Reply:
Got it.